x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/operator/arithmetic/Arithmetics.java

```diff
@@ -97,10 +97,10 @@ public static Number div(Number l, Number r) {
             return l.floatValue() / r.floatValue();
         }
         if (l instanceof Long || r instanceof Long) {
-            return l.longValue() / r.longValue();
+            return (double)l.longValue() / (double)r.longValue();
         }
 
-        return l.intValue() / r.intValue();
+        return (double)l.intValue() / (double)r.intValue();
```

Comment:
These are the functions that are currently implemented at QL level. The original `div` implementation didn't match EQL  implementation. EQL converts numbers into floats which will render a different results.
```
82/2 == 41
83//2 == 41
```
while 
```
82.0/2.0 == 41
82.0/2.0 == 41.5
```
Do we match the QL `div` function or do we introduce something else?

Reply:
We either need to change the spec for what `divide` and `/` mean, or we create a new function and update org/elasticsearch/xpack/eql/plugin/eql_whitelist.txt to use our version of `div`

Reply:
The operator classes (`DefaultBinaryArithmeticOperation`) work with any kind of number as they are type aware using the Java/SQL semantics
If the numbers are integers, so will the result - if a higher precision is needed then at least one of the numbers has to be a float/double.
You can check `ArithmeticOperation.dataType()` method. Clearly `+`, `-` and `*` preserve the type but `/` might be an exception to this rule.

Making `/` always return a float means it can turn an entire expression into a float: ` 2 /1 +1 = 3.0` or maybe `2.9999...`
Hence why currently it forces the users to think of types which works in SQL but not in EQL.
An alternative would be to have a dedicated `DIV` implementation for EQL.

Making this a team-discuss.

Reply:
Per our Slack conversation rolling back this change, allowing `div` implementation to deviate from the original EQL implementation. 
Expecting the type conversion to be explicit with ES EQL implementation for now.